osu.Game/Screens/Ranking/Expanded/Statistics/PerformanceStatistic.cs

```diff
@@ -22,30 +29,91 @@ public class PerformanceStatistic : StatisticDisplay
 
         private RollingCounter<int> counter;
 
+        [Resolved]
+        private ScorePerformanceCache performanceCache { get; set; }
+
+        [Resolved]
+        private BeatmapDifficultyCache difficultyCache { get; set; }
+
+        [Resolved]
+        private BeatmapManager beatmapManager { get; set; }
+
         public PerformanceStatistic(ScoreInfo score)
             : base("PP")
         {
             this.score = score;
         }
 
         [BackgroundDependencyLoader]
-        private void load(ScorePerformanceCache performanceCache)
+        private void load()
         {
-            if (score.PP.HasValue)
+            Task.WhenAll(
+                // actual performance
+                performanceCache.CalculatePerformanceAsync(score, cancellationTokenSource.Token),
+                // performance for a perfect play
+                getPerfectPerformance(score)
+            ).ContinueWith(attr =>
             {
-                setPerformanceValue(score.PP.Value);
-            }
-            else
+                PerformanceAttributes[] result = attr.GetResultSafely();
+                setPerformanceValue(new PerformanceBreakdown { Performance = result[0], PerfectPerformance = result[1] });
+            });
```

Comment:
not super sure all of this logic should reside in a drawable such as this, especially so that the data is just shipped down to the tooltip. would probably consider splitting off a separate component to compute this - could end up needing to reuse it later in other contexts, too.